core/GlobalState.cc

```diff
@@ -1121,12 +1131,11 @@ SymbolRef GlobalState::enterNewMethodOverload(Loc sigLoc, SymbolRef original, co
     return res;
 }
 
-SymbolRef GlobalState::enterFieldSymbol(Loc loc, SymbolRef owner, NameRef name) {
-    ENFORCE(owner.isClassOrModule(), "entering field symbol into not-a-class");
+SymbolRef GlobalState::enterFieldSymbol(Loc loc, ClassOrModuleRef owner, NameRef name) {
     ENFORCE(name.exists());
 
     auto flags = Symbol::Flags::FIELD;
-    SymbolData ownerScope = owner.dataAllowingNone(*this);
+    SymbolData ownerScope = SymbolRef(owner).dataAllowingNone(*this);
```

Comment:
I didn't add a `dataAllowingNone` method to `ClassOrModuleRef`, but maybe I should add one before merging. 🤔 

I think this method is primarily useful for a `ClassOrModuleRef`.

Reply:
Done.